system/middleware.go

```diff
@@ -41,6 +42,45 @@ func (application *Application) ApplyDbMap(c *web.C, h http.Handler) http.Handle
 	return http.HandlerFunc(fn)
 }
 
+func (application *Application) ApplyAPIAuth(c *web.C, h http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		if c.Env["IsAPI"].(bool) {
+			var apitoken string
+
+			authHeader := r.Header.Get("Authorization")
+			if strings.Index(authHeader, "Bearer ") == 0 {
```

Comment:
was going to suggest that too, but for readability

This is in a few spots I think.